public/pages/DetectorDetail/containers/DetectorDetail.tsx

```diff
@@ -256,22 +304,22 @@ export const DetectorDetail = (props: DetectorDetailProps) => {
           </EuiFlexItem>
         </EuiFlexGroup>
       </EuiFlexGroup>
-      {showDeleteDetectorModal ? (
+      {detecorDetailModel.showDeleteDetectorModal ? (
         <EuiOverlayMask>
           <ConfirmModal
             title="Delete detector?"
             description="Detector and feature configuration will be removed and the deletion is irreversible and you can’t review anomaly result of this detector on Kibana. To confirm deletion, click the delete button."
             callout={monitorCallout}
             confirmButtonText="Delete"
             confirmButtonColor="danger"
```

Comment:
We could still simplify this as mentioned in previous comment. We can set the state with all common props so that we just have to write single ConfirmMoal ?

Reply:
Have put all of current state into`DetectorDetailModel`. 
Tried to reduce to only one `ConfirmModal`. I need to add much more logic to generate ConfirmModal props. The code will be hard to maintain. I prefer to keep some redundant code to make the code logic more clear.